compiler/ETA/CodeGen/Expr.hs

```diff
@@ -319,3 +341,11 @@ cgAlgAltRhss binder alts = do
       allBranches = [ (getDataConTag con, code)
                     | (DataAlt con, code) <- taggedBranches ]
   return (maybeDefault, branches)
+
+cgTick :: Tickish Id -> CodeGen ()
+cgTick (SourceNote srcSpan _) = do
+  let srcLoc = realSrcSpanStart srcSpan
+  traceCg (str $ "StgTick , tickish: SourceNote, srcSpan: " ++ show srcSpan)
+  addLineNumber $ srcLocLine srcLoc
+  setSourceFileName $ fastStringText $ srcLocFile srcLoc
```

Comment:
`setSourceFileName` should ideally do some sort of filtering. Because according to this logic, the last SourceNote in a given binding will determine the SourceFile and I don't think this will always be the current module (due to inlining).